### PR TITLE
DSD-871: Pagination bug fix, adds test

### PR DESCRIPTION
**Pull request by oliviawongnyc**
Fixes JIRA ticket [DSD-871](https://jira.nypl.org/browse/DSD-871).

## This PR does the following:

- Fixed bug where the Next button in `Pagination` would navigate to the previous page.
- Added a test for Previous and Next link functionality.

## How has this been tested?

- Tested locally in the Test App, and added a unit test.

## Accessibility concerns or updates

- None.

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr908-eahdydi1c6vrjxhdeooo1ibqdrqwfoyv.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/62389c28fa61ec8fbf18245a

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ Currently, this repo is in Prerelease. When it is released, this project will ad
 - Fixes the border and padding from the `fieldset` HTML element by removing it.
 - Fixes alignment for long labels in the `Checkbox` and `Radio` components.
 - Fixes the line height for long labels in the `Toggle` component.
+- Fixes bug where the Next button in `Pagination` would navigate to the previous page.
```
Comment (EdwinGuzman) [inline CHANGELOG.md]: This should be moved to a new `### Fixes` section under the `## Prerelease` section (which will be for 0.25.13).

**Review by EdwinGuzman — COMMENTED** [review]
The update looks good but the changelog should be updated.

**Comment by oliviawongnyc** [comment]
> The update looks good but the changelog should be updated.

Updated!

**Review by EdwinGuzman — APPROVED** [review]

**Pull request merged**